### ISSUE TITLE
Query: Invalid include path failures

### ISSUE THREAD

**Issue by smitpatel** [dotnet/efcore#16090]
```C#
[ConditionalTheory]
        [MemberData(nameof(IsAsyncData))]
        public virtual Task String_include_multiple_derived_collection_navigation_with_same_name_and_different_type(bool isAsync)
        {
            var expectedIncludes = new List<IExpectedInclude>
            {
                new ExpectedInclude<InheritanceDerived1>(e => e.CollectionDifferentType, "CollectionDifferentType"),
                new ExpectedInclude<InheritanceDerived2>(e => e.CollectionDifferentType, "CollectionDifferentType")
            };

            return AssertIncludeQuery<InheritanceBase1>(
                isAsync,
                i1s => i1s.Include("CollectionDifferentType"),
                expectedIncludes);
        }
```
Exception
```
Message: 
    System.InvalidOperationException : Invalid include path: 'CollectionDifferentType' - couldn't find navigation for: 'CollectionDifferentType'
  Stack Trace: 
    at NavigationExpandingVisitor.TryConvertToLambdaInclude(MethodCallExpression methodCallExpression) in NavigationExpandingVisitor_MethodCall.cs line: 999
    at NavigationExpandingVisitor.ProcessInclude(MethodCallExpression methodCallExpression) in NavigationExpandingVisitor_MethodCall.cs line: 1016
    at NavigationExpandingVisitor.VisitMethodCall(MethodCallExpression methodCallExpression) in NavigationExpandingVisitor_MethodCall.cs line: 96
    at MethodCallExpression.Accept(ExpressionVisitor visitor)
    at NavigationExpander.ExpandNavigations(Expression expression) in NavigationExpander.cs line: 25
    at QueryOptimizer.Visit(Expression query) in QueryOptimizingExpressionVisitor.cs line: 26
    at QueryCompilationContext.CreateQueryExecutor[TResult](Expression query) in QueryCompilationContext2.cs line: 59
    at Database.CompileQuery[TResult](Expression query, Boolean async) in Database.cs line: 72
    at QueryCompiler.CompileQueryCore[TResult](IDatabase database, Expression query, IModel model, Boolean async) in QueryCompiler.cs line: 108
    at <>c__DisplayClass9_0`1.<Execute>b__0() in QueryCompiler.cs line: 97
    at CompiledQueryCache.GetOrAddQueryCore[TFunc](Object cacheKey, Func`1 compiler) in CompiledQueryCache.cs line: 84
    at CompiledQueryCache.GetOrAddQuery[TResult](Object cacheKey, Func`1 compiler) in CompiledQueryCache.cs line: 59
    at QueryCompiler.Execute[TResult](Expression query) in QueryCompiler.cs line: 93
    at EntityQueryProvider.Execute[TResult](Expression expression) in EntityQueryProvider.cs line: 79
    at EntityQueryable`1.GetEnumerator() in EntityQueryable`.cs line: 94
    at List`1.ctor(IEnumerable`1 collection)
    at Enumerable.ToList[TSource](IEnumerable`1 source)
    at QueryAsserter`1.AssertIncludeQuery[TItem1](Func`2 actualQuery, Func`2 expectedQuery, List`1 expectedIncludes, Func`2 elementSorter, List`1 clientProjections, Boolean assertOrder, Int32 entryCount, Boolean isAsync, String testMethodName) in QueryAsserter.cs line: 853
    at ComplexNavigationsQuerySqlServerTest.String_include_multiple_derived_collection_navigation_with_same_name_and_different_type(Boolean isAsync) in ComplexNavigationsQuerySqlServerTest.cs line: 3767
    at --- End of stack trace from previous location where exception was thrown ---
```

**Comment by maumar** [comment]
dupe of #15381

**Comment by ajcvickers** [comment]
This issue was closed as a dupe of #15381 which is now closed, but there are still failing tests closed against this issue.

**Comment by ajcvickers** [comment]
@smitpatel to follow up.
